docs/DeveloperGuide.md

```diff
@@ -1,33 +1,212 @@
 # Developer Guide
+##Content 
+[to be filled] 
+
+## Setting up, Getting Started
+###Setting up the project in your computer
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. Configure the JDK: Configuring the JDK to ensure Intellij is configured to use JDK 11.
+2. Import the project as a Gradle project.
+3. Run the studyit.StudyIt and try a few commands.
+4. Run the tests to ensure they all pass.
+
+###Before writing code
+
+1. Configure the coding style. 
+Ensure Coding style is standardised throughout. 
+2. Learn the design. 
+When you are ready to start coding, we recommend that you get some 
+sense of the overall design by reading about AddressBook’s architecture.
 
 ## Design & implementation
+###Architecture
+[to be filled]
+###Bookmark Component
+
+This bookmark section consists of how the bookmark feature is implemented. 
+The bookmark feature is implemented similarly to the main architecture, however, 
+in a smaller scale and a more bookmark-specific way. Figure 1 illustrates the general overview, 
+the associations and the multiplicity of the bookmark classes.
+
+![Bookmark Class Diagram](Images/BookmarkRun_Class.png)
+
+API: java.bookmarkRun
+
+The bookmark component consists of six major classes: `BookmarkRun`, `BookmarkStorage`, `BookmarkUi`, 
+`BookmarkCategory`, `BookmarkList` and `BookmarkParser`. 
+As shown in figure 1, `BookmarkRun` is 
+* The main class to be called when the bookmark mode is accessed.
+* The main class to access other bookmark classes.
+
+Given below, Figure 2, is the sequence diagram of how the classes interact with each 
+other when bookmark mode is accessed from the main function.
+
+![Bookmark Sequence Diagram](Images/sequence_bookmark.png)
+
```

Comment:
There are some problems in this diagram.
- Overlapping of method calls.
- Placement of conditional in alt block is not correct.
- I am unsure of the use of the dotted line that cuts horizontally through alt block.
- Self invocation is not drawn correctly.
- Some actuation bars extends after the method returns a value.

Reply:
![Screenshot 2020-10-28 at 11 41 18 AM](https://user-images.githubusercontent.com/57558909/97388173-810ab380-1912-11eb-8471-868f9945ec3f.png)
Here are more details.